### ISSUE TITLE
Integrate Connect Calendar feature

### ISSUE THREAD

**Issue by maxceem** [topcoder-platform/taas-app#519]
Recently we've implemented Connect Calendar feature in API in this challenge https://www.topcoder.com/challenges/af9a1041-7a78-484d-87d3-5aec9e87b578?tab=details

Now we would like to integrate it with UI.

To work on this task you would have to run locally TaaS API and TaaS App as per [this guide](https://github.com/topcoder-platform/taas-app/wiki/Run-TaaS-App-with-TaaS-API-locally-to-test-Interview-Scheduling-using-Nylas).

#### General Logic

- User clicks a link to connect calendar
- Redirected to Google/Microsoft/Nylas
- From there it redirected to TaaS API to save connected calendar details
- And from TaaS API it is redirected back to TaaS APP
- When redirected to TaaS App from link we have to know what interview popup to open and if calendar connection was successful or failed.

#### Task 

![image](https://user-images.githubusercontent.com/146016/138262357-4271d3d5-a816-4bb7-b2f4-dc33c62edb35.png)

- Build link as described in this challenge https://www.topcoder.com/challenges/af9a1041-7a78-484d-87d3-5aec9e87b578?tab=details
   - As `redirectTo` set URL to the current page + add `?interviewWithCandidate={jobCandidateId}`, example `https://platform.topcoder-dev.com/taas/myteams/18907/positions/a14d4946-418c-4329-998e-918603348fe2/candidates/interviews?interviewWithCandidate=8b0346e8-76bd-4c8c-b378-d27d3ce24262`
- When calendar is connected and redirected back to TaaS APP there could be 2 situations:
   - **calendar connected successfully**:
      - in this case it would redirect to the page with the next params `?interviewWithCandidate={jobCandidateId}&calendarConnected=true`
     - show interview scheduling modal as if we click button "schedule interview" https://monosnap.com/file/jzsr7LVUjWxz5lv2Yb7pKppXKG3wiR
     - show green toast which say that "Calendar was successfully connected."
     - remove part `?interviewWithCandidate={jobCandidateId}&calendarConnected=true` from URL
   - **calendar connection failed**:
       - in this case it would redirect to the page with the next params `?interviewWithCandidate={jobCandidateId}&calendarConnected=false&error={errorReason}`
       - show **red** toast which say that "Failed to connect calendar: {errorReason}"
       - remove part `?interviewWithCandidate={jobCandidateId}&calendarConnected=false&error={errorReason}` from URL
              
#### Verification

Provide demo video(s) showing both cases: calendar connected successfully or failed.

**Comment by deblasis** [comment]
Hi @maxceem! I am happy to work on this if you agree

**Comment by maxceem** [comment]
Sure, assigned you @deblasis.

**Comment by sdgun** [comment]
@mahidulalvi-bonic @maxceem Seeing the below error when trying to connect the calender


https://user-images.githubusercontent.com/4592712/139621417-cf97c8c1-413b-4005-8b50-0d1b1f739314.mp4

**Comment by sandhiyakavi** [comment]
@maxceem While trying the same above scenario, I am getting a different error "{"message":"Could not verify JWT token."}"
at the api level. 
![image](https://user-images.githubusercontent.com/55895538/139628054-2836d166-22d5-4d10-af49-b986ac17b744.png)

**Comment by maxceem** [comment]
Thank you @sdgun @sandhiyakavi 

We would try to debug it. If happen to have such issues again, could you please, copy/paste the URL from the browser which you had when error happens. It might help debugging it.

**Comment by sandhiyakavi** [comment]
@maxceem tried again, Getting TokenExpiredError with url https://api.topcoder-dev.com/v5/taas-teams/user-meeting-settings/callback?state=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI3MWM1ZTZhOC01MWQ5LTRmYjUtOTFjZS1kOTc0NjQyNTMxYWYiLCJyZWRpcmVjdFRvIjoiaHR0cHM6Ly9wbGF0Zm9ybS50b3Bjb2Rlci1kZXYuY29tL3RhYXMvbXl0ZWFtcy8xODk2Mi9wb3NpdGlvbnMvYTY4MTUyNTUtNWQ4MC00NGIwLTkzMmItODM4OWU4Njc1NjBiL2NhbmRpZGF0ZXMvdG8tcmV2aWV3P2ludGVydmlld1dpdGhDYW5kaWRhdGU9YjZjMTMwNjMtM2ZiYy00OWY0LTgzMTYtMWVkZGQ5NDVlOTYwIiwiaWF0IjoxNjM1NzYwODc2LCJleHAiOjE2MzU3NjA5MzZ9.wnk1WShZ9GaWGu_gPLRg9qLJ61UhRxb0_hsLALsKHtI&code=wVs8SPRgvEk7nGXX6DkE7o1Ck2Z7dr

![image](https://user-images.githubusercontent.com/55895538/139655480-43d4dcf7-6d57-4c7c-8ee1-a28d1b4cf00f.png)

**Comment by maxceem** [comment]
@sdgun regarding your issue during connecting calendar. I see that you are using email `taasintake3@gmail.com`. I wonder if you every used the same email with `+` sign like `taasintake3+something@gmail.com`?

I've noticed that if we use with Nylas email with `+` then if we try to connect calendar with the same email with `+` or without `+` it got broken.

So we always have to use real different emails we cannot use the same email with `+` because it points to the same Google account and lead to such issues.

cc @sandhiyakavi just in case

**Comment by maxceem** [comment]
@sandhiyakavi could you, please, try testing it one more time. Your issue has to be resolved now.

**Comment by sdgun** [comment]
@maxceem So far I only had this issue with taasintake3 which I have used with a + to create another user. I will check with some more users tommorrow.

**Comment by sandhiyakavi** [comment]
@maxceem Verified. I am not getting TokenExpiredError now.

**Comment by sdgun** [comment]
@maxceem Tried with a fresh user today and I am seeing these errors


https://user-images.githubusercontent.com/4592712/139788948-e7bbfc78-df1b-4fb0-8a51-01005a559d03.mp4

**Comment by maxceem** [comment]
@sdgun does this error happen if you select calendar permissions from the first time without going back?

**Comment by sdgun** [comment]
@maxceem Yes

**Comment by sdgun** [comment]
@maxceem Don't see any of the errors seen in the above comment video anymore, but the first time a new user connecting calender, always got the below error and it didn't connect the calender. But it worked second time.

Error: Could not find any writable calender


https://user-images.githubusercontent.com/4592712/140027082-6dcbe21f-9719-4a40-a32e-60ee31056629.mp4

**Comment by mahidulalvi-bonic** [comment]
Hi @maxceem,  just saw this, I encountered this issue when creating new users in Exchange and it used to go away after some time but usually happened just after creating Exchange account.

**Comment by maxceem** [comment]
@mahidulalvi-bonic I did some debugging and it looks likes a Nylas issue. At the moment when it redirects us to API the calendar is not yet returned by Nylas API. But if we try to get it after some time it’s there. I’ve reported it to Nylas support and waiting for response.

**Comment by sdgun** [comment]
> Thank you @sdgun @sandhiyakavi
> 
> We would try to debug it. If you happen to have such issues again, could you please, copy/paste the URL from the browser which you had when error happens. It might help debugging it.

I got this sporadically again, here is the url:

https://api.nylas.com/oauth/callback?state=7zfb3l5qmnreyoft6kb55j76i&code=4/0AX4XfWiRH2Mn7Iqvhr69FYBwAxubQZTa-Ry4LL1xmAuvUGcFJmFsS4bPaJMQye-0oD-6zA&scope=email%20profile%20https://www.googleapis.com/auth/calendar%20openid%20https://www.googleapis.com/auth/userinfo.profile%20https://www.googleapis.com/auth/userinfo.email&authuser=4&prompt=consent


https://user-images.githubusercontent.com/4592712/140510573-7a77a306-1f3e-4f12-b25c-4a85c7f738f4.mp4

**Comment by maxceem** [comment]
Thank you for the demo video @sdgun. By any chance did you ever use the same email with `+` sign? like `taasintake11+somthing@gmail.com`.

**Comment by sdgun** [comment]
@maxceem As far as I remember I haven't used it

**Comment by sdgun** [comment]
@maxceem We still have https://github.com/topcoder-platform/taas-app/issues/519#issuecomment-958729774, I guess this has not been addressed yet

**Comment by maxceem** [comment]
> @maxceem We still have #519 (comment), I guess this has not been addressed yet

Yeah, it still has to be fixed. I created a separate issue to track it https://github.com/topcoder-platform/taas-app/issues/580.

Is this the only one thing which left not working here?

**Comment by sdgun** [comment]
@maxceem We are waiting for connect Microsoft calender testing, if it is handled in a separate ticket that's all the issues for this

**Comment by maxceem** [comment]
@sdgun got it. I've created a separate issue for Microsoft calendar for easier tracking, so it's clear what has to be done https://github.com/topcoder-platform/taas-app/issues/581

**Comment by sdgun** [comment]
Removing the milestone since this feature was disabled in the current production release https://github.com/topcoder-platform/taas-app/issues/598
